e2e/nomostest/nt.go

```diff
@@ -581,8 +583,8 @@ func (nt *NT) ValidateMetricNotFound(metricName string) error {
 }
 
 // ValidateReconcilerErrors validates that the `reconciler_error` metric exists
-// for the correct reconciler and the tagged component has the correct value.
-func (nt *NT) ValidateReconcilerErrors(reconcilerName, component string) error {
+// for the correct reconciler pod and the tagged component has the correct value.
+func (nt *NT) ValidateReconcilerErrors(podName, component string) error {
```

Comment:
Since the consumer effectively has to always call `nt.GetDeploymentPod` before calling `nt.ValidateReconcilerErrors`, could we accept `reconcilerName` in this function and handle the mapping (i.e. call `nt.GetDeploymentPod`) in this function?

Reply:
never really liked this wrapper with a switch statement and hard coded strings... I just replaced it with a `ValidateReconcilerErrors(reconcilerName string, sourceCount, syncCount int)` that does the pod lookup for you. Now GetDeploymentPod is only used in 3 places.